pkg/controller/sub_controller/disaggregated_cluster/computegroups/prepare_modify.go

```diff
@@ -0,0 +1,237 @@
+	// get user and password
+	secret, _ := k8s.GetSecret(ctx, k8sclient, cluster.Namespace, cluster.Spec.AuthSecret)
+	adminUserName, password := resource.GetDorisLoginInformation(secret)
+
+	// get host and port
+	// When the operator and dcr are deployed in different namespace, it will be inaccessible, so need to add the dcr svc namespace
+	host := cluster.GetFEVIPAddresss()
+	confMap := dcgs.GetConfigValuesFromConfigMaps(cluster.Namespace, resource.FE_RESOLVEKEY, cluster.Spec.FeSpec.ConfigMaps)
+	queryPort := resource.GetPort(confMap, resource.QUERY_PORT)
+
+	// connect to doris sql to get master node
+	// It may not be the master, or even the node that needs to be deleted, causing the deletion SQL to fail.
+	dbConf := mysql.DBConfig{
+		User:     adminUserName,
+		Password: password,
+		Host:     host,
+		Port:     strconv.FormatInt(int64(queryPort), 10),
+		Database: "mysql",
+	}
+	// Connect to the master and run the SQL statement of system admin, because it is not excluded that the user can shrink be and fe at the same time
+	masterDBClient, err := mysql.NewDorisMasterSqlDB(dbConf)
+	if err != nil {
+		klog.Errorf("getMasterSqlClient NewDorisMasterSqlDB failed, get fe node connection err:%s", err.Error())
+		return nil, err
+	}
+	return masterDBClient, nil
+}
+
+// isDecommissionProgressFinished check decommission status
+// if not start decommission or decommission succeed return true
+func (dcgs *DisaggregatedComputeGroupsController) decommissionProgressCheck(masterDBClient *mysql.DB, cgName string, cgKeepAmount int32) (resource.DecommissionPhase, error) {
+	allBackends, err := masterDBClient.GetBackendsByCGName(cgName)
+	if err != nil {
+		klog.Errorf("decommissionProgressCheck failed, ShowBackends err:%s", err.Error())
+		return resource.DecommissionPhaseUnknown, err
+	}
+	dts := resource.ConstructDecommissionTaskStatus(allBackends, cgKeepAmount)
+	return dts.GetDecommissionPhase(), nil
+}
+
+func getScaledOutBENode(
+	masterDBClient *mysql.DB,
+	cgName string,
+	cgKeepAmount int32) ([]*mysql.Backend, error) {
+
+	allBackends, err := masterDBClient.GetBackendsByCGName(cgName)
+	if err != nil {
+		klog.Errorf("scaledOutBEPreprocessing failed, ShowBackends err:%s", err.Error())
+		return nil, err
+	}
+
+	var dropNodes []*mysql.Backend
+	for i := range allBackends {
+		node := allBackends[i]
+		split := strings.Split(node.Host, ".")
+		splitCGIDArr := strings.Split(split[0], "-")
+		podNum, err := strconv.Atoi(splitCGIDArr[len(splitCGIDArr)-1])
+		if err != nil {
+			klog.Errorf("scaledOutBEPreprocessing splitCGIDArr can not split host : %s,err:%s", node.Host, err.Error())
+			return nil, err
+		}
+		if podNum >= int(cgKeepAmount) {
+			dropNodes = append(dropNodes, node)
+		}
+	}
+	return dropNodes, nil
+}
+
+func skipApplyStatefulset(err error) bool {
+	if err == nil || err.Error() == decommissioningMessage {
```

Comment:
use message judge is not stable, if some add a space or update a char, this judge is failed.